### ISSUE TITLE
When will the code be released

### ISSUE THREAD

**Issue by DeepRunner** [ZihaoWang-CV/CAMP_iccv19#1]
Hello，when will the code be released？ Thanks

**Comment by ZihaoWang-CV** [comment]
Thanks for your interest in our paper and sorry for the delayed release date!

Since the code is based on some internal distributed libs of Sensetime, I'm currently rebuilding and testing the code with PyTorch. The code and models would be released hopefully in 2 weeks.

**Comment by zl535320706** [comment]
keep in focus!

**Comment by czhxiaohuihui** [comment]
The method for this paper is actually similar to Dynamic Fusion with Intra- and Inter-modality Attention Flow for Visual Question Answering. In Image-Text Matching, I think the main challenge is pair-to-pair feature fusion, for example, if the batch size is 128 (128 image-text pairs), there will be 128*128 aggregated feature. It's time-consuming and calculation-comsuming!!!  How do you solve this problem?

**Comment by shuchenweng** [comment]
It's been two weeks. Have you finished debugging your code?

**Comment by ZihaoWang-CV** [comment]
> The method for this paper is actually similar to Dynamic Fusion with Intra- and Inter-modality Attention Flow for Visual Question Answering. In Image-Text Matching, I think the main challenge is pair-to-pair feature fusion, for example, if the batch size is 128 (128 image-text pairs), there will be 128*128 aggregated feature. It's time-consuming and calculation-comsuming!!! How do you solve this problem?

Sorry for my late response!

Since the image features are pre-extracted from the Faster R-CNN, it only costs around 12GB/GPU memories and 1.1s/batch while distributed running on 16 TITAN XP GPUs. (or ~28GB/GPU, 0.8s/batch on 8 V100 GPUs).

Our code is based on the implementation of VSE++ and SCAN, we compute 128*(128/num_devices) pairs on each GPU and gather the final score matrix(128*128*1) on GPU 0 to compute the final loss. This could be simply implemented by expanding the text feature with shape (batch_size, word_num, dim) to (device_num, batch_size, word_num, dim) and passing it through the nn.DataParallel.

The code would be released as soon as approved by the information security management group of the company. I think the security review could be done in this week.

**Comment by czhxiaohuihui** [comment]
Thanks a lot for your response!

**Comment by flauted** [comment]
Is there an update on this?

**Comment by ZihaoWang-CV** [comment]
> Is there an update on this?

Sorry again for the delayed release date. The security review has not been finished yet. But after some discussion, I could release part of the implementation and the pre-trained model weights later.
 
I have left CUHK and Sensetime several month ago and the email address has been deprecated. Please directly contact zihaowang.cv@gmail.com or submit a new issue if you have further questions. Thank you!
